src/main/java/uk/gov/pay/products/resources/ProductResource.java

```diff
@@ -183,4 +232,17 @@ public Response findProductsAndStats(@QueryParam("gatewayAccountId") Integer gat
         List<ProductUsageStat> usageStats = productFactory.productFinder().findProductsAndUsage(gatewayAccountId);
         return Response.status(OK).entity(usageStats).build();
     }
+
+    private String getDescription(Product product) {
+        switch (product.getType()) {
+            case ADHOC:
+                return String.format("Token for \"%s\" payment link", product.getName());
+            case PROTOTYPE:
+                return String.format("Token for Prototype: %s", product.getName());
+            case DEMO:
+                return "Token for Demo Payment";
+            default:
+                return null;
```

Comment:
We should throw an exception if we reach here: something has gone terribly wrong.